.github/workflows/test.yml

```diff
@@ -0,0 +1,235 @@
+# [Required secrets]
+#   UNITY_LICENSE: Contents of an Unity license file (*.ulf)
+name: Test
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - "**/Dockerfile"
+      - .github/workflows/test.yml
+      - reference-project-test/*
+  pull_request:
+    types:
+      - "opened"
+      - "synchronize"
+    paths:
+      - "**/Dockerfile"
+      - .github/workflows/test.yml
+      - reference-project-test/*
```

Comment:
What's the rationale about becoming so specific about the triggers?

For example; why not simply trigger on all default PR events?

```yml
on:
  pull_request: {}
```

Reply:
> What's the rationale about becoming so specific about the triggers?

This is to prevent unnecessary testing due to changes in files that are not related to building the image (e.g. changing only `Readme.md`).
The test (48 jobs) takes about 1 h.
No problem?

Reply:
I'd say lets keep the paths, but remove the specific types. What do you think?

Reply:
`workflow_dispatch` and `push` may be redundant.
(They support the following)
- Trigger from other workflows or manually
- When a change is pushed directly

Reply:
> I'd say lets keep the paths, but remove the specific types. What do you think?

OK

Reply:
Removed.

Reply:
I think we should keep `push` and `workflow_dispatch `.
These will work in the forked repository, so we can test before creating a PR.